proto/cosmos/tx/v1beta1/tx.proto

```diff
@@ -254,3 +262,7 @@ message AuxSignerData {
   // sig is the signature of the sign doc.
   bytes sig = 4;
 }
+
+// This is an empty message to be added as a response for failed messages
+// when dealing with non-atomic multi-msg txs.
+message MsgFailureResponse {}
```

Comment:
will the user know which tx failed and why it failed through other means?

Reply:
Which msg failed, but the index. Why it failed, by looking at the events. But it may be a good idea to include something like `errorsmod.Wrapf(err, "failed to execute message; message index: %d", i)` inside MsgFailureResponse

Reply:
i think its better to return the error, its easy for us to do in order to reduce overhead for users

Reply:
+1, the proto field can be a string.

Also, we have a `cosmos.msg.v1` package. Maybe that's a better place to put this struct?